### PR TITLE
buildtest.yml: force sparse warnings as errors

### PR DESCRIPTION
**Pull request by plbossart**
Without this sparse warnings are silently ignored

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>

### CODE REVIEW

**Review by kv2019i — DISMISSED** [review]
Some existing errors will need cleanup, but now they are at least noticed.

File: .github/workflows/buildtest.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@ jobs:
 
       - name: build start
         run: |
-          export ARCH=x86_64
+          export ARCH=x86_64 CF="-Wsparse-error"
           bash kconfig/kconfig-sof-default.sh
           make modules_prepare
           make M=sound/soc/sof C=2
```
Comment (marc-hb) [inline .github/workflows/buildtest.yml]:
```suggestion
          make -k M=sound/soc/sof C=2
```

**Review by marc-hb — COMMENTED** [review]
> Some existing errors will need cleanup, but now they are at least noticed.

Only if they're fixed very soon, otherwise people get used to them very quickly, stop looking at them and miss any new ones.

File: .github/workflows/buildtest.yml (hunk 1 above)
Comment (plbossart) [reply .github/workflows/buildtest.yml]: done

**Comment by plbossart** [comment]
with today's fixes this should work now. I have a set of patches coming to raise the bar with more warnings and other directories.

**Review by marc-hb — APPROVED** [review]

**Pull request merged**